### PR TITLE
[Entrypoints][Refactor] Refactor llumnix entrypoints to be more modular

### PR DESCRIPTION
**Pull request by s5u13b**
Previously, we invasively added a lot of codes to the api server of llm engine to configure the llumnix environments in entrypoints. We refactor llumnix entrypoints codes in this pr. With this pr, we can configure the llumnix environments by adding a small amount of codes to the api server of llm engine.

### CODE REVIEW

File: llumnix/config/default.py [hunk 1]
```diff
@@ -21,34 +21,35 @@
...
+# Queue type for request output queue
+_C.SERVER.QUEUE_TYPE = "rayqueue"
+# Port number for the request output queue
+_C.SERVER.REQUEST_OUTPUT_QUEUE_PORT = 1234
 # Disable logging requests in server
 _C.SERVER.DISABLE_LOG_REQUESTS_SERVER = False
 # Enable logging request timestamp
 _C.SERVER.LOG_REQUEST_TIMESTAMPS = False
+# Config file of Llumnix arguments
+_C.SERVER.CONFIG_FILE = None
```
Comment (KuilongCui) [inline llumnix/config/default.py:44]: There is no need to set CONFIG_FILE in yaml

File: llumnix/arg_utils.py [hunk 2]
```diff
@@ -22,6 +22,77 @@
...
+    queue_type: str = None
+    request_output_queue_port: int = None
+    disable_log_requests_server: bool = None
+    log_request_timestamps: bool = None
+    config_file: bool = None
+
+    def __post_init__(self):
+        for attr in dataclasses.fields(self):
+            if getattr(self, attr.name) is None:
+                setattr(self, attr.name, getattr(_C.SERVER, attr.name.upper()))
```
Comment (KuilongCui) [inline llumnix/arg_utils.py:59]: add a  checker for the default value must be None

File: llumnix/entrypoints/utils.py [hunk 3]
```diff
@@ -27,21 +27,37 @@
 from llumnix.utils import random_uuid
 from llumnix.arg_utils import EngineManagerArgs
 from llumnix.queue.queue_type import QueueType
+from llumnix.server_info import ServerInfo, RequestTimestamps
+from llumnix.queue.utils import init_output_queue_server
 
-logger = init_logger(__name__)
+logger = init_logger("llumnix.api_server")
```
Comment (KuilongCui) [inline llumnix/entrypoints/utils.py:33]: why change

File: llumnix/entrypoints/vllm/api_server.py [hunk 4]
```diff
@@ -11,125 +11,51 @@
...
+                                            manager_abort,
+                                            manager_is_ready)
 from llumnix.logger import init_logger
 from llumnix.utils import random_uuid
-from llumnix.backends.vllm.utils import check_engine_args
-from llumnix.queue.queue_server_base import QueueServerBase
-from llumnix.queue.utils import get_output_queue_server
 from llumnix.config import get_llumnix_config, LlumnixConfig
 
 logger = init_logger("llumnix.api_server")
```
Comment (KuilongCui) [inline llumnix/entrypoints/vllm/api_server.py:42]: init_logger("llumnix.api_server") -> init_logger("llumnix.entrypoints.vllm.api_server")  maybe better

**Review by KuilongCui — APPROVED** [review]

File: llumnix/config/default.py (hunk 1 above)
Comment (s5u13b) [reply llumnix/config/default.py:44]: why?

File: llumnix/arg_utils.py (hunk 2 above)
Comment (s5u13b) [reply llumnix/arg_utils.py:59]: Get

File: llumnix/entrypoints/utils.py (hunk 3 above)
Comment (s5u13b) [reply llumnix/entrypoints/utils.py:33]: Does the logger can take effects when 'logger = init_logger(__name__)'?

File: llumnix/entrypoints/vllm/api_server.py (hunk 4 above)
Comment (s5u13b) [reply llumnix/entrypoints/vllm/api_server.py:42]: Get

File: llumnix/arg_utils.py [hunk 5]
```diff
@@ -22,6 +22,77 @@
...
+        if self.cur_namespace == 'llumnix' and "--help" not in args:
+            assert 'default' not in kwargs or kwargs['default'] is None, \
+                f"Do not set the default value for '{args[0]}' in CLI, or set default value to None. " \
+                f"The default value will be retrieved from config/default.py in get_llumnix_config."
+            if kwargs.get('action') == 'store_true':
+                kwargs['default'] = None
+        super().add_argument(*args, **kwargs)
+
+
+# All the default values of llumnix arguments are setted in default.py. So all the arguments here are setted to None.
```
[the remark this answers is not in the document]
Comment (s5u13b) [reply llumnix/arg_utils.py:44]: Get

File: llumnix/entrypoints/utils.py [hunk 6]
```diff
@@ -170,35 +191,110 @@ def init_llumlets(engine_manager_args: EngineManagerArgs, engine_args, node_id:
...
                                   available_instance_ids, available_llumlets)
         logger.info("Init Llumnix components done, {} instances are ready, instance_ids: {}."
                     .format(len(available_instance_ids), available_instance_ids))
 
-    return engine_manager, available_instance_ids, available_llumlets
+    request_output_queue = init_output_queue_server(ip, request_output_queue_port, request_output_queue_type)
+
+    return engine_manager, available_instance_ids, available_llumlets, request_output_queue
+
+def setup_llumnix(engine_manager_args, engine_args, cfg):
```
[the remark this answers is not in the document]
Comment (s5u13b) [reply llumnix/entrypoints/utils.py:227]: MakeEntrypointContext is also hard to be understand for me. Actually, I think llumnix_entrypoints_context is not a very good name, but I cannot find better one.

File: llumnix/entrypoints/utils.py (hunk 3 above)
Comment (KuilongCui) [reply llumnix/entrypoints/utils.py:33]: you can give it a try :)

File: llumnix/config/default.py (hunk 1 above)
Comment (KuilongCui) [reply llumnix/config/default.py:44]: I think this is a chicken or egg situation...

File: llumnix/entrypoints/utils.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (zhypku) [reply llumnix/entrypoints/utils.py:227]: it seems to me that all this function does is making a context (from your return value)

**Review by zhypku — APPROVED** [review]

**Pull request merged**